server/src/main/java/org/apache/druid/server/coordinator/loading/CuratorLoadQueuePeon.java

```diff
@@ -72,7 +72,7 @@ public class CuratorLoadQueuePeon implements LoadQueuePeon
    * with loading or dropping segments.
    */
   private final ExecutorService callBackExecutor;
-  private final DruidCoordinatorConfig config;
+  private final Duration loadTimeout;
 
   private final AtomicLong queuedSize = new AtomicLong(0);
```

Comment:
This class along with its test can be removed right? I see:

> The code for ZK-based segment loading is not being removed yet as it might be needed for testing purposes.

Are we referring to the unit tests? Is there a plan to migrate them to use the http strategy since it's the only option now?

Reply:
We need not migrate the `LoadQueuePeonTest` as there is already an `HttpLoadQueuePeonTest`.
There are some other tests though, such as `DruidCoordinatorTest` which will need to be migrated to use the `HttpLoadQueuePeon`.

> > The code for ZK-based segment loading is not being removed yet as it might be needed for testing purposes.

> Are we referring to the unit tests?

No, I wanted to retain it for now just in case we need to do an implementation/perf comparison between the ZK and HTTP implementation, or if there is an issue and we want to re-enable ZK.

Druid 31 will have the option removed. Druid 32 will have the code removed completely as well.